extensions/table.c

```diff
@@ -155,13 +219,49 @@ static table_row *row_from_string(cmark_syntax_extension *self,
       node_cell *cell = (node_cell *)parser->mem->calloc(1, sizeof(*cell));
       cell->buf = cell_buf;
       cell->start_offset = offset;
-      cell->end_offset = offset + cell_matched - 1;
+      if (cell_matched > 0)
+        cell->end_offset = offset + cell_matched - 1;
+      else
+        cell->end_offset = offset;
 
       while (cell->start_offset > 0 && string[cell->start_offset - 1] != '|') {
         --cell->start_offset;
         ++cell->internal_offset;
       }
 
+      cell->cell_data = (node_cell_data *)parser->mem->calloc(1, sizeof(node_cell_data));
+
+      // Check for a column-spanning cell
+      if (row->n_columns > 0 && cmark_strbuf_len(cell->buf) == 0 && cell->start_offset == cell->end_offset) {
+        cell->cell_data->colspan = 0;
+
+        // find the last cell that isn't part of a colspan, and increment that colspan
+        cmark_llist *tmp = row->cells;
+        node_cell *colspan_cell = NULL;
+        while (tmp) {
+          node_cell *this_cell = (node_cell *)tmp->data;
+          if (this_cell->cell_data->colspan > 0)
+            colspan_cell = this_cell;
+          tmp = tmp->next;
+        }
+        if (colspan_cell)
+          ++colspan_cell->cell_data->colspan;
+      } else {
+        cell->cell_data->colspan = 1;
+      }
+
+      // Check for a row-span marker. Actually incrementing the spanning cell's rowspan will happen later
```

Comment:
Super nit, but I read this "Actually" sentence as "the previous statement isn't true. Consider dropping the "Actually" to IMO reduce ambiguity.

Reply:
```
// Check this cell for a row-span marker, so the spanning cell can have its rowspan incremented later.
```

What about this?